src/main/java/ca/on/oicr/gsi/migration/PinerySource.java

```
@@ -459,6 +481,54 @@ private void connectHierarchy(Collection<MigrationSample> samples, Collection<Mi
       fixDilutionAttributes(dilution);
     }
   }
+
+  private MigrationSample deduplicateSample(Map<String, MigrationSample> ghostSamples, MigrationSample sample) {
+    String key = GhostUtils.generateAlias(sample);
+    MigrationSample existing = ghostSamples.get(key);
+    // if it's not in the map, return add it and return the sample
+    if (existing == null) {
+      ghostSamples.put(key, sample);
+      existing = sample;
+    } else {
+      // if it is in the map, check if the sample class is the same
+      // if not, increment the sibling number and call this method again
+      if (sample.getSampleClass() != existing.getSampleClass()) {
+        Integer inc = sample.getSiblingNumber();
+        if (inc == null) {
+          String regex = "(.*)(\\d+)$";
+          Matcher m = Pattern.compile(regex).matcher(key);
+          if (!m.matches()) {
+            log.error("Cannot determine previous number in order to increment for sample:" + key);
+            return null;
```

Should throw an exception to fail migration - either ConversionException, or more likely just a RuntimeException since it should never actually happen. For the match to fail, there would have to be an error in the `GhostUtils.generateAlias` code